src/main/kotlin/com/valensas/ftp/model/SFTPClient.kt

```diff
@@ -56,6 +55,19 @@ class SFTPClient : FTPClient() {
             throw e
         }
 
+    override fun storeFile(
+        remote: String?,
+        local: InputStream?,
+    ): Boolean {
+        try {
+            channel.put(local, remote)
+            return true
+        } catch (e: Exception) {
+            logger.error("Error happened while uploading file", e)
+            throw e
+        }
+    }
+
     fun listFilesAtPath(pathname: String): List<LsEntry> {
```

Comment:
listeleme fonksiyonunu da bir şekilde ortaklaştırmamız lazım. Bizim kütüphane ve FtpClient kütüphanesi farklı class tipleri dönüyor ondan aklıma gelen listFileNameAtPath(ismi farklı olabilir) diye bir fonksiyon yazıp tüm classlarda pathdeki dosyaların ismini ve size'ını dönmek.  Jobs servisinde FtpService classında listFiles fonksiyonunda iki class içide bu yapılıyor oradan bakılabilir. 

 fun listFiles(remotePath: String): List<Pair<String, Long>> {
        try {
            if (ftpClient is SFTPClient) {
                return ftpClient.listFilesAtPath(remotePath).map {
                    it.filename to it.attrs.size
                }
            }
            return ftpClient.listFiles(remotePath).map {
                it.name to it.size
            }
        } catch (e: Exception) {
            logger.error("Can not list files at $remotePath, ", e)
            throw e
        }
    }

Reply:
Tamamdır, ekleyeceğim @KaanSimsek19

Reply:
Bunu ekledim @KaanSimsek19

Reply:
Bu tam istediğim gibi olmamış şimdi listFilesAtPath any döndürüyor yani bu kütüphaneyi kullanan servislerde yine clientın tipine bakıp ona göre işlem yapmamız gerekicek. Bence en temizi ayrı bir fonksiyonda listeleme işlemi yapıp sadece isim ve size ikililerini dönmek. Listelemeyi kullanan yerlerde zaten sadece bu ikisi kullanılıyor dönen dosyalarda. Eğer diğer bilgiler gerekirse zaten listFiles ve listFilesAtPath methodu durduğu için kullanılabilir.

fun listFileInfo() { // sftp için
ftpsClient.listFilesAtPath(remotePath).map {
it.filename to it.attrs.size
}
}

fun listFileInfo() { // ftp için
ftpClient.listFiles(remotePath).map {
it.name to it.size
}
}

Reply:
Tamamdır. Söylediğin şekilde güncelledim @KaanSimsek19